ddtrace/opentracer/tracer.py

```diff
@@ -235,14 +237,18 @@ def start_span(
             scope = self._scope_manager.active
             parent_span = getattr(scope, "span", None)
             ot_parent_context = getattr(parent_span, "context", None)
-            # we want the ddcontext of the active span in order to maintain the
-            # ddspan hierarchy
-            dd_parent = getattr(ot_parent_context, "_dd_context", None)
-
-            # if we cannot get the context then try getting it from the DD tracer
-            # this emulates the behaviour of tracer.trace()
-            if dd_parent is None:
-                dd_parent = self._dd_tracer.get_call_context()
+
+            # Compare the active ot and dd spans. Using the one which
+            # was created later as the parent.
```

Comment:
Yeah - unfortunately this is a somewhat unrelated to the main context change but I think it was necessary due to some tests breaking.

Before, if there was both an opentracing span active (managed by the opentracing context management) _and_ a datadog span active (from the ddtrace context management) we always opted to take the datadog span. To me this didn't make sense because consider the case where the autoinstrumentation is provided by ddtrace and manual tracing is provided by opentracing.


```python
# django integration
with tracer.trace("django.request", ...) as dd_root:
    with opentracer.start_active_span() as ot_parent:
        with opentracer.start_active_span() as ot_child:
            ...
```

before: `ot_child` would be a child of `dd_root`

after: `ot_child` would be a child of the most recently created active span which is `ot_parent`.

Reply:
Yeah I do briefly mention it but it's a good idea to elaborate a little bit. Probably also worth a release note as a fix.